tests/test_click.py

```diff
@@ -0,0 +1,307 @@
+        button.addEventListener('dblclick', event => {
+          window.double = true;
+        });
+    """
+                     )
+    button = await p.J('button')
+    await button.click(clickCount=2)
+    assert await p.evaluate('double') == True
+    assert await p.evaluate('result') == 'Clicked'
+
+
+@sync
+async def test_click_partially_obscured_button(isolated_page, server):
+    p = isolated_page
+    await p.goto(server / '/input/button.html')
+    await p.evaluate("""
+    () => {
+        const button = document.querySelector('button');
+        button.textContent = 'Some really long text that will go offscreen';
+        button.style.position = 'absolute';
+        button.style.left = '368px';
+    }
+    """)
+    await p.click('button')
+    assert await p.evaluate('() => window.result') == 'Clicked'
+
+
+@sync
+async def test_click_rotated_button(isolated_page, server):  # line 215 in pup
+    p = isolated_page
+    await p.goto(server / '/input/rotatedButton.html')
+    await p.click('button')
+    assert await p.evaluate('() => result') == 'Clicked'
+
+
+@sync
+async def test_right_click_contextmenu_event(isolated_page, server):
+    p = isolated_page
+    await p.goto(server / '/input/scrollable.html')
+    await p.click('#button-8', button='right')
+    assert await p.evaluate(
+        "() => document.querySelector('#button-8').textContent"
+    ) == 'context menu'
+
+
+@sync
+async def test_click_links_causing_navigation(isolated_page, server):
+    p = isolated_page
+    await p.setContent(
+        f"""<a href = "{server.empty_page}" > empty.html </a>"""
+    )
+    # this await should not hang
+    await p.click('a')
+
+
+@sync
+async def test_click_button_inside_iframe(isolated_page, server):
+    p = isolated_page
+    await p.goto(server.empty_page)
+    await p.setContent('<div style="width:100px;height:100px">spacer</div>')
+    await attachFrame(p, server / '/input/button.html', 'button-test')
+    frame = p.frames[1]
+    button = await frame.J('button')
+    await button.click()
+    assert await frame.evaluate("() => window.result") == 'Clicked'
+
+
+@sync
+@pytest.mark.skip('skipped in upstream puppeteer')
+async def test_fixed_button_inside_iframe(isolated_page, server):
+    p = isolated_page
+    await p.goto(server.empty_page)
+    await p.setViewport(dict(width=500, height=500))
+    await p.setContent('<div style="width:100px;height:2000px">spacer</div>')
+    await attachFrame(p, server.cross_process_server / '/input/button.html', 'button-test')
+    frame = p.frames[1]
+    await frame.evaluate(
+        'button',
+        "button => button.style.setProperty('position', 'fixed')"
+    )
+    await frame.click('button')
+    assert await frame.evaluate("() => window.result") == 'Clicked'
+
+
+@sync
+async def test_click_button_devicescalefactor(isolated_page, server):
+    p = isolated_page
+    await p.setViewport(dict(width=500, height=500, deviceScaleFactor=5))
```

Comment:
I'd prefer a dict literal here
```suggestion
    await p.setViewport({'width': 500, 'height': 500, 'deviceScaleFactor': 5})
```